app/models/category.rb

```diff
@@ -0,0 +1,9 @@
+class Category < ApplicationRecord
+  has_many :category_bills, dependent: :destroy
+  has_many :bills, through: :category_bills
+  belongs_to :user, class_name: 'User', foreign_key: 'user_id'
+
+  has_one_attached :icon
+
+  validates :name, presence: true
+end
```

Comment:
- [ ] I noticed that in you seem to have missed validating the icon and user_id columns. Kindly include validations for these column as well.